prime-angular-frontend/src/app/shared/pipes/care-setting-abbr.pipe.ts

```diff
@@ -0,0 +1,32 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { CareSetting } from '@enrolment/shared/models/care-setting.model';
+import { CareSettingEnum } from '@shared/enums/care-setting.enum';
+
+@Pipe({
+  name: 'careSettingAbbr'
+})
+export class CareSettingAbbrPipe implements PipeTransform {
+
+  transform(value: number | number[]): string {
+    if (Array.isArray(value)) {
+      return value.map(item => this.getAbbr(item)).join(', ');
+    }
+    return this.getAbbr(<CareSettingEnum>value);
+  }
+
+  private getAbbr(careSetting: CareSettingEnum) {
+    switch (careSetting) {
+      case CareSettingEnum.COMMUNITY_PHARMACIST:
+        return 'CP';
+      case CareSettingEnum.DEVICE_PROVIDER:
+        return 'DP';
+      case CareSettingEnum.HEALTH_AUTHORITY:
+        return 'HA';
+      case CareSettingEnum.PRIVATE_COMMUNITY_HEALTH_PRACTICE:
+        return 'PCHP';
+      default:
+        return '-';
+    }
```

Comment:
From the OOP perspective, seems these abbr strings can be part of the CareSettingEnum rather than defined in a helper transform class. Any benefit of using the PipeTransform?

Reply:
In the enum we already map the code to the name. As far as I'm aware in TS we can't map a third property. The other thing is we can't define a helper method to map these abbr, without merging the enum with a namespace. I went with this route as i didn't really want to clutter the enum, but im open to some suggestions as maybe containing the infor to that file might be more beneficial.

Reply:
> In the enum we already map the code to the name. As far as I'm aware in TS we can't map a third property. The other thing is we can't define a helper method to map these abbr, without merging the enum with a namespace. I went with this route as i didn't really want to clutter the enum, but im open to some suggestions as maybe containing the infor to that file might be more beneficial.

Fair enough. LGTM. 

Another possible way is to have static instances of a regular class CareSetting, so that the third property can be added.  


```
class CareSetting {
    id: number;
    name: string;
    abbr: string;
    
    public static COMMUNITY_PHARMACIST: CareSetting = new CareSetting(0, 'COMMUNITY_PHARMACIST', 'CP');
    .......
}
```

Reply:
Just pushed some changes that move the abbr logic into the enum. With not a crazy amount of magic, seems like the way to merge a enum and namespace

Reply:
> Just pushed some changes that move the abbr logic into the enum. With not a crazy amount of magic, seems like the way to merge a enum and namespace

Agreed that they belong to be in the same file, makes more sense and easier reading as well..